### PR TITLE
ci: add dummy jobs to required workflows

### LINKED ISSUE #2890: Add dummy jobs to required workflows
### Describe the feature

We have some required checks for PRs/merges but currently GitHub only supports adding required checks *per job* which is really hard to maintain.

For example, our unit testing workflow has a matrix named `test (m/n)`. In order to require that this jobs passes, we have to individually add `test (1/4)`, `test (2/4)` and so on. If we ever want to change the size of this matrix to e.g. `n = 5` or `n = 3`, we have to go back and update the required checks.

As a workaround (for now) we should add a single `{workflow name}-success` job at the end of workflows that simply mark the workflow as succeeded. This job should depend on all other jobs we require to complete for the workflow to be considered a success.

Our required checks would only be these `-success` jobs. This makes it easier to tune our CI without having to also change our required checks.

For example, for the `bench.yml` workflow, this job would be added:

```yml
bench-success:
  needs: [bench-check]
  steps:
    - run: echo OK.
```

## How to contribute

I marked this as https://github.com/paradigmxyz/reth/labels/D-good-first-issue. In order to contribute, the following jobs need to be adjusted:

image

Once the PR is opened and approved, we will change over the required checks before merging. The PR itself will be in a perpetual pending state until then, since the current required checks will not be run.

### Additional context

Different types of "required before merging" protections are planned, see e.g. https://github.com/orgs/community/discussions/46757#discussioncomment-4909850

### PR DESCRIPTION
**Pull request by TechieBoy**
Closes #2890

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -51,3 +51,9 @@ jobs:
       - uses: Swatinem/rust-cache@v2
       - name: Check if documentation builds
         run: RUSTDOCFLAGS="-D warnings" cargo doc --all --no-deps --all-features --document-private-items
+
+  lint-success:
+    name: lint success
+    needs: [code-lint, doc-lint]
```
Comment (onbjerg) [inline .github/workflows/ci.yml:57]: I think this needs to be `[lint, doc-lint]`?

**Review by onbjerg — CHANGES_REQUESTED** [review]

**Review by onbjerg — APPROVED** [review]
seeing if this will trigger ci :shrug:

**Review by onbjerg — CHANGES_REQUESTED** [review]
ah, each of the jobs also needs `runs-on: ubuntu-latest`, my bad

**Comment by TechieBoy** [comment]
Looks like it worked. All the success ones were triggered in correct order

**Review by onbjerg — APPROVED** [review]

**Pull request merged**